python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -624,6 +634,47 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
         return []
 
 
+@whitelist_for_serdes
+class SkipOnRequiredButNonexistentParentsRule(
+    AutoMaterializeRule, NamedTuple("_SkipOnRequiredButNonexistentParentsRule", [])
+):
+    @property
+    def decision_type(self) -> AutoMaterializeDecisionType:
+        return AutoMaterializeDecisionType.SKIP
+
+    @property
+    def description(self) -> str:
+        return "parents are required but do not exist"
```

Comment:
maybe something that references partitions here would be a bit more informative. I was thinking something along the lines of

"required parent partitions do not exist"

The only concern there is just that people might confuse "not exist" with "have not been materialized" -- curious if you have thoughts on a better term / phrase there

Reply:
I think this description looks clear to me.

I don't think people would be confused about this language, but we can also add a more verbose tooltip (i.e. `required upstream partitions are not defined in upstream asset X`)